src/Utilities/StdHelpers.hpp

```diff
@@ -307,41 +307,41 @@ inline std::string current_date_and_time() {
 
 // Arithmetic operators for std::array<T, Dim>
 
-template <size_t Dim, typename T>
+template <size_t Dim, typename T, typename U>
 inline std::array<T, Dim>& operator+=(
     std::array<T, Dim>& lhs,
-    const std::array<T, Dim>& rhs) noexcept(noexcept(lhs[0] += rhs[0])) {
+    const std::array<U, Dim>& rhs) noexcept(noexcept(lhs[0] += rhs[0])) {
   for (size_t i = 0; i < Dim; ++i) {
     gsl::at(lhs, i) += gsl::at(rhs, i);
   }
   return lhs;
 }
 
-template <size_t Dim, typename T>
+template <size_t Dim, typename T, typename U>
 inline std::array<T, Dim> operator+(
```

Comment:
What if T is an int and U is a double? Then the return type is wrong.

Reply:
It's worse than this even. Initially we realised this and said to use auto return type along with a `std::array<decltype(lhs[0]+rhs[0]),Dim>`, but if either of T or U is a DataVector then decltype(lhs[0] + rhs[0]) returns some Blaze internal type. I don't think we'd settled on an easy way to fix this.

Reply:
Good point. We probably want this to evaluate eagerly, right? Or at least don't care about building up expressions, I think. In which case you could do:

```
auto operator+() -> std::array<
    tmpl::conditional_t<cpp17::is_same_v<T, DataVector> or
                                   cpp17::is_same_v<U, DataVector>,
                                   DataVector,
                                   decltype(std::decval<T>() + std::declval<U>())>, Dim>
```

What do you guys think about that? The only thing I dislike about this is that it would mean whenever this code lives needs to know about DataVector.  This could be solved by adding overloads to DataVector.hpp that handles DataVector and U, as well as T and DataVector. That would be the least messy way, I think.

Reply:
By the way, these types of notes should be added as open questions in the description. I think I haven't been very good at this, but it's something we should all work on. The goal would just be to avoid forgetting a discussion on an edge case.